src/Microsoft.FeatureManagement/FeatureManagementBuilder.cs

```diff
@@ -37,15 +37,33 @@ public FeatureManagementBuilder(IServiceCollection services)
 
             if (!Services.Any(descriptor => descriptor.ServiceType == serviceType && descriptor.ImplementationType == implementationType))
             {
-                Services.AddSingleton(serviceType, implementationType);
+                //
+                // Register the feature filter with the same lifetime as the feature manager
+                if (!Services.Any(descriptor => descriptor.ServiceType == typeof(FeatureManager) && descriptor.Lifetime == ServiceLifetime.Scoped))
```

Comment:
Are we registering `FeatureManager` as ServiceType or ImplementaionType? I'm expecting descriptor.ServiceType to be `typeof(IFeatureManager)`

Reply:
Yes, it should be IFeatureManager, Jimmy has called out this in this PR #303. I did not update it in this branch.

Reply:
Updated.